ferry/deploy.py

```diff
@@ -185,61 +220,100 @@ def _match(name):
         revert = conn.begin()
         conn.execute("SET CONSTRAINTS ALL DEFERRED;")
 
-        for table in meta.sorted_tables[::-1]:
-            if not table.name.endswith("_staged"):
-                print(f"Reverting table {table.name}")
-                conn.execute(
-                    f'ALTER TABLE "{table.name}" RENAME TO {table.name}_staged;'
-                )
-                conn.execute(f'ALTER TABLE "{table.name}_old" RENAME TO {table.name};')
+        for table in alchemy_tables:
+            print(f"Reverting table {table.name}")
+            conn.execute(
+                f'ALTER TABLE IF EXISTS "{table.name}" RENAME TO {table.name}_staged;'
+            )
+            conn.execute(
+                f'ALTER TABLE IF EXISTS "{table.name}_old" RENAME TO {table.name};'
+            )
 
         revert.commit()
         raise
 
+    # -----------------
+    # Remove old tables
+    # -----------------
+
     print("\n[Deleting temporary old tables]")
-    meta.reflect()  # update the models to reflect postgres
     delete = conn.begin()
     conn.execute("SET CONSTRAINTS ALL DEFERRED;")
-    for table in meta.sorted_tables[::-1]:
-        if table.name.endswith("_old"):
-            print(f"Dropping table {table.name}")
-            conn.execute(f"DROP TABLE IF EXISTS {table.name} CASCADE;")
+    for table in alchemy_tables:
+        print(f"Dropping table {table.name}_old")
+        conn.execute(f"DROP TABLE IF EXISTS {table.name}_old CASCADE;")
 
     delete.commit()
 
+    # ---------------
+    # Add constraints
+    # ---------------
+
     # add back the foreign key constraints
-    print("\n[Regenerating constraints]")
+    print(f"\n[Regenerating constraints ({len(constraints)})]")
     regen_constraints = conn.begin()
     for constraint in constraints:
         if isinstance(constraint, ForeignKeyConstraint):
```

Comment:
I believe we have a couple other constraints too - is there a reason to limit to just ForeignKeyConstraints?

Reply:
The other constraints are primary keys and checks, which are handled by the table creation process previously I think (will check on this)